pilot/api/data.py

```diff
@@ -271,13 +293,17 @@ def transfer(self, files, activity='default', **kwargs):  # noqa: C901
             activity.append('default')
 
         copytools = None
+        current_activity = None
```

Comment:
Wen, why you confuse and complicate the logic, what did you want to cover? -- just ensure that `activity` name is exist in `astorages`?

Reply:
That's just one thing. I think when selecting one copytool, the corresponding activity must be remembered. The activity even should be remembered before copytools. 

I also need to find the corresponding storage for the copytool later.
https://github.com/wguanicedew/pilot2/blob/next/pilot/api/data.py#L637
For example, a site is configured with {'pr': ['BNL_OSG2_DATADISK'], 'es_events':['CERN_PROD_ES']} and {'pr':['gfalcopy'], 'es_events':['objectstore']}. I set the the stageout activity to be 'es_events' in the es jobs(https://github.com/wguanicedew/pilot2/blob/next/pilot/eventservice/workexecutor/plugins/genericexecutor.py#L180). At first it tries to use 'objectstore' to stageout to BNL_OSG2_DATADISK. That's not what I want.

Reply:
see my comment below: again, instead of patching workflow, do pass properly filled FileSpec entries.